actix-settings/src/lib.rs

```diff
@@ -258,14 +260,20 @@ where
     B: MessageBody + 'static,
 {
     fn apply_settings(mut self, settings: &ActixSettings) -> Self {
-        if settings.tls.enabled {
-            // for Address { host, port } in &settings.actix.hosts {
-            //     self = self.bind(format!("{}:{}", host, port))
-            //         .unwrap(/*TODO*/);
-            // }
-            unimplemented!("[ApplySettings] TLS support has not been implemented yet.");
-        } else {
-            for Address { host, port } in &settings.hosts {
+        for Address { host, port } in &settings.hosts {
+            #[cfg(feature = "tls")]
+            {
+                if settings.tls.enabled {
+                    self = self.bind_openssl(format!("{}:{}", host, port), settings.tls.get_ssl_acceptor_builder())
+                        .unwrap(/*TODO*/)
```

Comment:
some TODOs here and a few lines down.

Perhaps split this into a function which returns a `Result<Self>`, and a deprecated function that returns `Self`

Reply:
I couldn't understand exactly what you meant, I think it's eliminating the `unwrap` and then split into functions? Correct me if I'm wrong or maybe I haven't understood your ideology.

Reply:
I mean revise the existing 

```diff
-fn apply_settings(mut self, settings: &ActixSettings) -> Self {
+fn try_apply_settings(mut self, settings: &ActixSettings) -> Result<Self, Error> {
```

and then append to the trait a new

```rust
#[deprecated]
fn apply_settings(mut self, settings: &ActixSettings) -> Self {
    self.try_apply_settings(settings).unwrap("try_apply_settings failed")
}
```

Reply:
Oh, I see, wouldn't that be better done in another PR?

Reply:
IMO this addition of `.unwrap(/*TODO*/)` warrants it being done in this PR.

previously there was only one .unwrap in this function, and it was for a very predictable and small problem set, typically port already being in use.

In your changes, there are now an additional source of potential panics, and the reasons why SSL init might cause panic are myriad.

Reply:
Ahhh ok, in a few hours, I'll take a look at this part and make these changes. Thanks.